index.js

```diff
@@ -37,10 +34,30 @@ var anymatch = function(criteria, value, returnIndex, startIndex, endIndex) {
     default:
       result = false;
     }
-    if (result) { matchIndex = index + startIndex; }
+    if (result) {
+      matchIndex = index + startIndex;
+    }
     return result;
   }
-  var matched = criteria.slice(startIndex, endIndex).some(testCriteria);
+  var crit = criteria;
+  var negGlobs = crit.reduce(function(arr, criterion, index) {
+    if (typeof criterion === 'string' && criterion[0] === '!') {
+      if (crit === criteria) {
+        // make a copy before modifying
+        crit = crit.slice();
+      }
+      crit[index] = null;
+      arr.push(criterion.substr(1));
+    }
+    return arr;
+  }, []);
+  if (!negGlobs.length || !micromatch.any(string, negGlobs)) {
+    if (path.sep === '\\' && typeof string === 'string') {
```

Comment:
I tried to do this while still being able to override the value for running the tests.

```js
    before(function() {
      path.sep = '\\';
      delete require.cache[require.resolve('./')];
      anymatch = require('./');
    });
```

But then I got errors like

```
  1) anymatch windows paths "before all" hook:
     Error: Cannot find module 'arrify'
      at Function.Module._resolveFilename (module.js:316:15)
      at Function.Module._load (module.js:258:25)
      at Module.require (module.js:345:17)
```

on io.js and node versions 0.11 & 0.12. Got the error even when changing the technique to use rewire or [require-uncached](https://github.com/sindresorhus/require-uncached). Node 0.10 worked.

Took me a while to figure out what was going wrong, but I finally realized that overriding `path.sep` was breaking `require.resolve` for npm packages in node_modules. Changing `require('arrify')` to `require('./node_modules/arrify')` fixed it, but breaking convention like that is not a worthwhile tradeoff.

The perf improvement being pursued here is really really negligible, so I'm just going to leave it as-is.